README.md

```diff
@@ -5,25 +5,89 @@
 
 You can view more platform specific details at [JETSON.md](./doc/JETSON.md) and [PI.md](./doc/PI.md).
 
 _Note: On a Raspberry Pi, you must install GStreamer plugins before running the module._
 
 _WARNING: There is a known issue for Debian Bookworm due to changes in the libcamerasrc plugin._
+___
 
-```bash
-sudo apt install libcamera0 gstreamer1.0-x gstreamer1.0-plugins-base gstreamer1.0-plugins-good gstreamer1.0-plugins-bad
-```
+## Usage
+
+To use this module, follow these instructions to [add a module from the Viam Registry](https://docs.viam.com/registry/configure/#add-a-modular-resource-from-the-viam-registry) and select the `viam:camera:csi` model from the [`csi-cam` module](https://app.viam.com/module/viam/csi-cam).
+
+## Configure your CSI camera
 
-See [registry-app-config.json](./etc/registry-app-config.json) for how to configure with csi-cam appimage from registry.
+> [!NOTE]  
+> Before configuring your camera, you must [create a robot](https://docs.viam.com/manage/fleet/robots/#add-a-new-robot).
 
+Navigate to the **Config** tab of your robot’s page in [the Viam app](https://app.viam.com/). Click on the **Components** subtab and click **Create component**. Select the `camera` type, then select the `csi` model. Enter a name for your camera and click **Create**.
+
+On the new component panel, copy and paste the following attribute template into your camera's **Attributes** box. 
+```json
+{
+  "width_px": <int>,
+  "height_px": <int>,
+  "frame_rate": <int>,
+  "debug": <bool>
+}
+```
+
+> [!NOTE]  
+> For more information, see [Configure a Robot](https://docs.viam.com/manage/configuration/).
+
+Edit the attributes as applicable and save your config.
+In the **Control** tab of the [Viam app](https://app.viam.com/), you can now view the camera feed. 
+If you do not see anything, check the logs tab for errors.
+
+### Attributes
+
+The following attributes are available for `viam:camera:csi` cameras:
+
+<!-- prettier-ignore -->
+| Name | Type | Inclusion | Description |
+| ---- | ---- | --------- | ----------- |
+| `width_px` | int | Optional | Width of the image this camera captures in pixels. <br> Default: `1920` |
+| `height_px` | int | Optional | Height of the image this camera captures in pixels. <br> Default: `1080` |
+| `frame_rate` | int | Optional | The image capture frame rate this camera should use. <br> Default: `30` |
+| `video_path` | string | Optional | The filepath to the input sensor of this camera on your board. If none is given, your robot will attempt to detect the video path automatically. <br> Default: `"0"` </br>  |
+| `debug` | boolean | Optional | Whether or not you want debug input from this camera in your robot's logs. <br> Default: `false` |
+
+### Example Configuration
+
+```json
+{
+  "modules": [
+    {
+      "executable_path": "/usr/local/bin/viam-csi",
+      "name": "csi_cam_module"
```

Comment:
Shouldn't it be this? Otherwise it's for a local module. Or do we want to document both?
```suggestion
      "type": "registry",
      "name": "viam_csi-cam",
      "module_id": "viam:csi-cam",
      "version": "0.2.0"
```

Reply:
Fine w/ just registry as points to local config file below

Reply:
Thank you!